exp/controllers/rosamachinepool_controller.go

```diff
@@ -123,7 +123,21 @@ func (r *ROSAMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	controlPlane := &rosacontrolplanev1.ROSAControlPlane{}
 	if err := r.Client.Get(ctx, controlPlaneKey, controlPlane); err != nil {
 		log.Info("Failed to retrieve ControlPlane from MachinePool")
-		return reconcile.Result{}, nil
+
+		if apierrors.IsNotFound(err) && rosaMachinePool.DeletionTimestamp != nil && !rosaMachinePool.DeletionTimestamp.IsZero() {
```

Comment:
```suggestion
		if apierrors.IsNotFound(err) && !rosaMachinePool.DeletionTimestamp.IsZero() {
```
no need to check for nil, `isZero()` does that.